### PR TITLE
samples: crypto: Add psa_tls .yaml configs

### PR DESCRIPTION
**Pull request by torsteingrindvik**
Make internal builds more one-to-one with public CI
by leveraging twister's support for building different
scenarios with different configs.

The scenarios will be combinations of:

* Server, client
* ECDSA, RSA
* Platforms
* Backend
	- On non-secure: Cryptocell
	- On secure: Cryptocell (legacy), oberon (legacy)

Ref.: NCSDK-13950

### CODE REVIEW

**Comment by Vge0rge** [comment]
The overlays look good, could we add the configs directory to place all the configuration files? Just to make the root directory a little bit more clean.

**Comment by torsteingrindvik** [comment]
> The overlays look good, could we add the configs directory to place all the configuration files? Just to make the root directory a little bit more clean.

Done

**Review by magnev — APPROVED** [review]

File: samples/crypto/psa_tls/overlays/oberon-legacy.conf [hunk 1]
```diff
@@ -0,0 +1,2 @@
+CONFIG_CC3XX_BACKEND=n
```
Comment (frkv) [inline samples/crypto/psa_tls/overlays/oberon-legacy.conf:1]: Unsure about negative prompt will exist for CONFIG_CC3XX_BAKEND. Please remove this

**Review by Vge0rge — APPROVED** [review]
LGTM

File: samples/crypto/psa_tls/overlays/oberon-legacy.conf (hunk 1 above)
Comment (torsteingrindvik) [reply samples/crypto/psa_tls/overlays/oberon-legacy.conf]: Removed, internal run still green and verified cc3xx was not turned on.

**Review by frkv — APPROVED** [review]
LGTM

**Pull request merged**